lib/helpers.ex

```diff
@@ -97,12 +97,23 @@ defmodule EVM.Helpers do
       iex> EVM.Helpers.encode_signed(-1)
       EVM.max_int() - 1
   """
-  @spec wrap_int(integer()) :: EVM.val
+  @spec encode_signed(integer()) :: EVM.val
   def encode_signed(n) when n < 0, do: EVM.max_int() - abs(n)
   def encode_signed(n), do: n
-  def decode_signed(n) do
-    <<sign :: size(1), _ :: bitstring>> = :binary.encode_unsigned(n)
-    if sign == 0, do: n, else: n - EVM.max_int()
+
+  @spec decode_signed(integer()) :: EVM.val
+  def decode_signed(n) when is_integer(n) do
+    decode_signed(:binary.encode_unsigned(n))
+  end
+
+  def decode_signed(n) when is_binary(n) do
```

Comment:
Just want to make sure we're safe with overflow issues here? Can we at least check that the byte_size is what we expect?

Reply:
We can!

Are we expecting a specific byte_size though? What should happen if we don't get that size? Return 0? `raise`? Happy to chat on Gitter for more context here...

Reply:
Well, it should only be in a truly exceptional case (since the rest of the code is supposed to check sizes), so it's likely just a simple assertion (e.g. it should raise). I'm less clear what the size is supposed to be, so we don't need to do it if it's non-obvious.